src/tokens/TokenBalances.tsx

```diff
@@ -127,6 +146,18 @@ function TokenBalancesScreen({ navigation }: Props) {
     })
   }
 
+  const handleMeasureBalanceHeight = (event: LayoutChangeEvent) => {
+    balanceHeightRef.current = event.nativeEvent.layout.height
+  }
+
+  const debouncedUpdateHeader = debounce((scrollOffset: number) => {
+    setShowBalanceInHeader(scrollOffset > balanceHeightRef.current)
+  }, HEADER_UPDATE_DEBOUNCE_TIME_MS)
+
+  const handleScroll = (event: { nativeEvent: NativeScrollEvent }) => {
+    debouncedUpdateHeader(event.nativeEvent.contentOffset.y)
```

Comment:
can't debounce at the `handleScroll` level because the synthetic scroll events are reused for performance reasons and would return null sometimes if debounced